test/plug/router_test.exs

```diff
@@ -317,6 +360,20 @@ defmodule Plug.RouterTest do
     end
   end
 
+  test "assigns path params to conn params" do
+    assert true == true
```

Comment:
(aside from the useless assert I cleaned up), your question is about why we are populating `conn.params`?

It grew out of the discussion here: https://github.com/elixir-lang/plug/issues/457